src/backend/cuda/jit.cpp

```diff
@@ -659,40 +407,12 @@ void evalNodes(vector<Param<T> >&outputs, vector<Node *> output_nodes)
     }
 
     for (int i = 0; i < num_outputs; i++) {
-        args.push_back(&outputs[i].ptr);
-    }
-
-    // DO NOT PUT THESE IN A SCOPE.
-    // The pointers are used later.
-    // Scoping them results in undefined behavior.
-
-    int strides[] = {(int)outputs[0].strides[0],
-                     (int)outputs[0].strides[1],
-                     (int)outputs[0].strides[2],
-                     (int)outputs[0].strides[3]};
-
-    int dims[] = {(int)outputs[0].dims[0],
-                  (int)outputs[0].dims[1],
-                  (int)outputs[0].dims[2],
-                  (int)outputs[0].dims[3]};
-
-    if (is_linear) {
-        int nelem = 1;
-        for (int i = 0; i < 4; i++) {
-            nelem *= outputs[0].dims[i];
-        }
-        args.push_back((void *)&nelem);
-    } else {
-        for (int i = 0; i < 4; i++) args.push_back((void *)(strides + i));
-        for (int i = 0; i < 4; i++) args.push_back((void *)(dims + i));
+        args.push_back((void *)&outputs[i]);
```

Comment:
We don't know the size of the args vector. It depends on what each type of the node is.